app/models/authentication.rb

```diff
@@ -19,8 +19,6 @@ def display_name
   protected
 
   def check_not_last
-    errors.add(:base, "Cannot delete an activated user's last authentication") \
-      if user.authentications.size == 1 && user.is_activated?
-    errors.none?
+    user.authentications.size > 1 || !user.is_activated?
```

Comment:
AR errors don't seem to work well outside of `validate` stuff.
For example, calling `authentication.valid?` would clear the error...
So I decided not to use them.